program/src/guards/freeze_token_payment.rs

```diff
@@ -164,7 +164,7 @@ impl Guard for FreezeTokenPayment {
             //   6. `[]` Metaplex Token Metadata program ID.
             FreezeInstruction::Thaw => {
                 msg!("Instruction: Thaw (FreezeTokenPayment guard)");
-                thaw_nft(ctx, route_context, data)
+                thaw_nft(ctx, route_context, data, false)
```

Comment:
Any reason we are disabling this for token payments? The payer will need some native SOLs anyway to pay for the transaction, right?

Reply:
Theres actually no reason to disable it other than the fact that its easier to not have to worry about that. Thinking about it closer there isnt a reason not to have it in for token payment too (the only issue is existing mints may need to fund that freeze pda account with some more sol to give out the crank reward. I think it might be better to just make it only take away that fee if the account is still going to be rent exempt after removing it and if not just assuming that the cm was created before this feature was added.

Reply:
@lorisleiva I have now made this work for the token payment too :)